msktutil.cpp

```diff
@@ -282,6 +282,11 @@ int finalize_exec(msktutil_exec *exec, msktutil_flags *flags)
     }
     VERBOSE("SAM Account Name is: %s", flags->sAMAccountName.c_str());
 
+    if (exec->mode == MODE_CREATE && !flags->use_service_account && flags->add_only_default_spns) {
+      exec->add_principals.push_back("host/" + flags->hostname);
+      exec->add_principals.push_back("host/" + get_short_hostname(flags));
```

Comment:
Does it actually make sense to restrict adding the short hostname to the host service? I'd suggest to just `push_back("host");` like in the original code, and modify `qualify_principal_vec()` to add both the long and the short hostname for any service.

Reply:
Adding the short hostname only to the host principal keeps changes visible to the user at a minimum. So yes, it makes sense. However, now, the default only works if `-s` is not specified. That would be a user visible change. Therefore sticking wich `push_back("host")` as in the original code does make sense. Hower I do not think `qualify_principal_vec` would be the best place...

Reply:
- `qualify_principal_vec` is a simple function that only adds `/hostname` if there ist no `/...` part in a principal.
- currently there is no special logic to add `/shortname` anywhere -- besides the keytab writing code in `msktkrb5.cpp` where it is should be removed for it only create a keytab entry without corresponding SPN.
- sticking with `push_back("host")` seems to be a good idea.
- but adding `host/shortname` only when no other service has been specified via `--server` feels wrong. So `add_only_default_spns` will not be necessary.

Reply:
I'm still concerned about different behaviour for computer and service accounts: For computer accounts, we add host SPNs for the full and short hostnames by default. For service accounts, we qualify based on an expliciat `--hostname` argument, which will typically be set to the full hostname. Now, if we consider the typical case of a separate service account for HTTP, we end up with an explicit SPN `HTTP/hostname` attached to the service account, and an implicit SPN `HTTP/short_hostname` attached to the computer account via AD's default `SPNMapping` rules. This is quite subtle, and therefore I'd favour more consistence in our UI, here.

Reply:
@mproehl Are you ok with the latest version of the PR?

Reply:
I would assume that no default SPNs are set with a service account. They need special, read explicit, treatment.